gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.java

```diff
@@ -2008,7 +2009,7 @@ public default GraphTraversal<S, E> filter(final Traversal<?, ?> filterTraversal
      *
      * @return the updated traversal with respective {@link NoneStep}.
      */
-    @Override
+    @Deprecated
```

Comment:
This deprecation needs javadoc, especially in this case. Here's an example of the form:

https://github.com/apache/tinkerpop/blob/master/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/Graph.java#L314-L323